lib/ui/survey_question/survey_questions_view_model.dart

```diff
@@ -58,13 +77,58 @@ class SurveyQuestionsViewModel extends StateNotifier<SurveyQuestionsViewState> {
   void nextQuestion() async {
     final index = _currentIndex.hasValue ? _currentIndex.value : 0;
     _currentIndex.add(index + 1);
+
+    if (_surveyQuestions.hasValue) {
+      _surveyNextQuestions.add(_surveyQuestions.value[index].displayType);
+    }
+  }
+
+  void saveAnswer(SubmitSurveyAnswerModel answer) async {
+    final questionId = _surveyQuestions.value[_currentIndex.value].id;
+    final existingIndex = _submitSurveyQuestions.indexWhere(
+      (question) => question.id == questionId,
+    );
+
+    if (existingIndex >= 0) {
+      _submitSurveyQuestions[existingIndex].answers.add(answer);
```

Comment:
The purpose of `existingIndex` is to handle updating the answer, isn't it? 🤔 Following the design, user doesn't allow to navigate back to the previous question so I think we could get rid of this.

Reply:
@hoangnguyen92dn, Could you please give me an example if we don't check `existingIndex` to map/define the index of `questionId` in the list 🙏

Reply:
```suggestion
  void saveAnswer(SubmitSurveyAnswerModel answer) async {
    final questionId = _surveyQuestions.value[_currentIndex.value].id;
    // final existingIndex = _submitSurveyQuestions.indexWhere(
    //   (question) => question.id == questionId,
    // );
    //
    // if (existingIndex >= 0) {
    //   _submitSurveyQuestions[existingIndex].answers.add(answer);
    // } else {
      _submitSurveyQuestions.add(
        SubmitSurveyQuestionModel(
          id: questionId,
          answers: [answer],
        ),
      );
    // }
  }
```

Sorry for the confusion if my comment above was not clear to you 😄 I meant that the user is only allowed to navigate to the next question, so we don't need to validate if the answer with the question is already cached, can we simply cache the answer with the current question instead?

Reply:
@hoangnguyen92dn, May I know Does it work properly on your side I got this error `{"errors":[{"detail":"Questions is invalid"}]}` when submitting the survey request, It seems the format is wrong 🤔 

```
flutter: *** DioError ***:
flutter: uri: https://survey-api.nimblehq.co/api/v1/responses
flutter: DioError [bad response]: The request returned an invalid status code of 422.
flutter: uri: https://survey-api.nimblehq.co/api/v1/responses
flutter: statusCode: 422
flutter: headers:
flutter:  connection: keep-alive
flutter:  cache-control: no-cache
flutter:  transfer-encoding: chunked
flutter:  date: Tue, 13 Jun 2023 02:14:45 GMT
flutter:  vary: Accept-Encoding, Origin
flutter:  content-encoding: gzip
flutter:  referrer-policy: strict-origin-when-cross-origin
flutter:  report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=ALsMv9Y3P6DXNmOaEoh7wS4y%2B2GbyDf5%2Bjf6Gkj3ziyBrLQ9pbdU9akpFkdLaam08P2sgBbyhlLpMuhS1VC2%2FDApvbncnpoUN7%2Fko1P%2FFx3b8y%2FkRI62dFbvtx4NRHUHcaSQqjv8640%2F"}],"group":"cf-nel","max_age":604800}
flutter:  cf-cache-status: DYNAMIC
flutter:  x-permitted-cross-domain-policies: none
flutter:  content-type: application/json; charset=utf-8
flutter:  x-xss-protection: 0
flutter:  server: cloudflare
flutter:  x-request-id: 8570c199-eca1-4536-ac0c-4861e6bc1f3c
flutter:  alt-svc: h3=":443"; ma=86400
flutter:  nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
flutter:  x-download-options: noopen
flutter:  cf-ray: 7d66d2653a4eee3d-BKK
flutter:  x-runtime: 0.022449
flutter:  via: 1.1 vegur
flutter:  x-frame-options: SAMEORIGIN
flutter:  x-content-type-options: nosniff
flutter: Response Text:
flutter: {"errors":[{"detail":"Questions is invalid"}]}
```

Reply:
@Wadeewee Ah I see your issue, so how about modifying the `saveAnswer` to accept the list of answers instead? It means we only call saveAnswer once before navigating to the next question.

```
  void saveAnswer(List<SubmitSurveyAnswerModel> answers) async {
    final questionId = _surveyQuestions.value[_currentIndex.value].id;
      _submitSurveyQuestions.add(
        SubmitSurveyQuestionModel(
          id: questionId,
          answers: answers,
        ),
      );
  }
```

Reply:
> how about modifying the saveAnswer to accept the list of answers instead?

Yup, It possible @hoangnguyen92dn by this way 🚀